google/cloud/bigquery/job.py

```diff
@@ -1871,6 +1870,12 @@ def to_api_repr(self):
         ]
 
         configuration = self._configuration.to_api_repr()
+        source_table = _helpers._get_sub_prop(
```

Comment:
I'm still very confused by this logic. I think it'll be unnecessary once https://github.com/googleapis/python-bigquery/pull/323 is merged.

Reply:
In the current situation i have to delete `sourceTable` from the already passed resource, else it throws an error like `can't have both sourceTable and sourceTables in config`. but it think after that PR merged don't need this type of check so i will wait for that PR to be merged and then update this PR accordingly.